lambda/translcore.ml

```diff
@@ -575,21 +747,40 @@ and transl_exp0 ~in_new_scope ~scopes e =
               of_location ~scopes e.exp_loc)
       end
   | Texp_ifthenelse(cond, ifso, Some ifnot) ->
-      Lifthenelse(transl_exp ~scopes cond,
-                  event_before ~scopes ifso (transl_exp ~scopes ifso),
-                  event_before ~scopes ifnot (transl_exp ~scopes ifnot),
-                  Typeopt.value_kind e.exp_env e.exp_type)
+      let k = value_kind_if_not_void e void_k in
+      Lifthenelse(transl_exp ~scopes None cond,
+                  event_before ~scopes ifso (transl_exp ~scopes void_k ifso),
+                  event_before ~scopes ifnot (transl_exp ~scopes void_k ifnot),
+                  k)
   | Texp_ifthenelse(cond, ifso, None) ->
-      Lifthenelse(transl_exp ~scopes cond,
-                  event_before ~scopes ifso (transl_exp ~scopes ifso),
+      Lifthenelse(transl_exp ~scopes None cond,
+                  event_before ~scopes ifso (transl_exp ~scopes void_k ifso),
```

Comment:
I agree `void_k` must be none here (but it seems cleaner to pass on the input than construct a new None?)

(I also have vague thoughts that it might be nice to allow `void` things in some of the positions where `unit` is today, like to the left of a semi-colon, or here.)